src/io/flutter/editor/FlutterSaveActionsManager.java

```diff
@@ -0,0 +1,195 @@
+    if (psiFile == null || !psiFile.isValid()) {
+      return;
+    }
+
+    final Module module = getModuleForPath(file);
+    if (module == null) {
+      return;
+    }
+
+    if (!DartPlugin.isDartSdkEnabled(module)) {
+      return;
+    }
+
+    // check for errors
+    if (PsiErrorElementUtil.hasErrors(myProject, psiFile.getVirtualFile())) {
+      return;
+    }
+
+    DartAnalysisServerService.getInstance(myProject).serverReadyForRequest(myProject);
+
+    if (settings.isOrganizeImportsOnSaveKey()) {
+      performOrganizeImportsOnSave(document, file);
+    }
+    else {
+      performFormatOnSave(document, file, false);
+    }
+  }
+
+  private void performOrganizeImportsOnSave(@NotNull Document document, @NotNull VirtualFile file) {
+    final String filePath = file.getPath();
+    final SourceFileEdit fileEdit = DartAnalysisServerService.getInstance(myProject).edit_organizeDirectives(filePath);
+    if (fileEdit != null) {
+      ApplicationManager.getApplication().invokeLater(() -> new WriteCommandAction.Simple(myProject) {
+        @Override
+        protected void run() {
+          boolean reSave = false;
+
+          if (AssistUtils.applyFileEdit(myProject, fileEdit)) {
+            // Committing a document here is required in order to guarantee that DartPostFormatProcessor.processText() is called afterwards.
+            PsiDocumentManager.getInstance(myProject).commitDocument(document);
+
+            reSave = true;
+          }
+
+          performFormatOnSave(document, file, reSave);
+        }
+      }.execute());
+    }
+  }
+
+  private void performFormatOnSave(@NotNull Document document, @NotNull VirtualFile file, boolean reSave) {
+    final int lineLength = getRightMargin(myProject);
+    final DartAnalysisServerService das = DartAnalysisServerService.getInstance(myProject);
+
+    das.updateFilesContent();
+
+    final DartAnalysisServerService.FormatResult formatResult = das.edit_format(file, 0, 0, lineLength);
+
+    if (formatResult == null) {
+      if (reSave) {
+        FileDocumentManager.getInstance().saveDocument(document);
+      }
+      return;
+    }
+
+    ApplicationManager.getApplication().invokeLater(() -> new WriteCommandAction.Simple(myProject) {
+      @Override
+      protected void run() {
+        boolean didFormat = false;
+
+        final List<SourceEdit> edits = formatResult.getEdits();
+        if (edits != null && edits.size() == 1) {
+          final String replacement = StringUtil.convertLineSeparators(edits.get(0).getReplacement());
+          document.replaceString(0, document.getTextLength(), replacement);
+          PsiDocumentManager.getInstance(myProject).commitDocument(document);
+
+          didFormat = true;
+        }
+
+        if (reSave || didFormat) {
+          FileDocumentManager.getInstance().saveDocument(document);
+        }
+      }
+    }.execute());
+  }
+
+  private @Nullable
+  Module getModuleForPath(@NotNull VirtualFile file) {
```

Comment:
getModuleForFile?

Reply:
Go To Symbol -> `findModuleForFile` will show you an existing helper method that works better :)

Reply:
Updated! Thanks :)